src/daml/Daml/Finance/Setup/Scenario/OtcSwaps.daml

```diff
@@ -66,10 +84,50 @@ run = do
 
   -- Instruments
-  usd   <- originateToken               centralBank centralBank pub epoch "USD"   "United States Dollar"
-  swap1 <- originateCreditDefaultSwap   seller      seller                "CDS1"  "Credit Default Swap 5Y"  pub (tt $ date 2020 tm td) (date 2020 tm td) ["EMPTY"] seller (date 2021 tm td) (date 2025 tm td) Act365Fixed Following 0.011 M 12 usd "DefaultProbability" "RecoveryRate" True
+  let
+    hibor6M = "HKD-HIBOR-6M"
+    hibor12M = "HKD-HIBOR-12M"
+    dpTencent = "DP-TENCENT"
+    rrTencent = "RR-TENCENT"
+  hkd  <- originateToken               cashProvider cashProvider pub epoch "HKD"   "Hong Kong Dollar"
+  irs1 <- originateInterestRateSwap    clearer      clearer                "IRS1"  "Interest Rate Swap 3Y 1.5%/HIBOR-6M"     pub (tt $ y 2022) (y 2022) ["EMPTY"] clearer (y 2023) (y 2025) Act365Fixed Following 0.025  M 6  hkd hibor6M  True
+  irs2 <- originateInterestRateSwap    clearer      clearer                "IRS2"  "Interest Rate Swap 5Y 1.75%/HIBOR-6M"    pub (tt $ y 2022) (y 2022) ["EMPTY"] clearer (y 2023) (y 2027) Act365Fixed Following 0.025  M 6  hkd hibor6M  True
+  irs3 <- originateInterestRateSwap    clearer      clearer                "IRS3"  "Interest Rate Swap 5Y 2.0%/HIBOR-12M"    pub (tt $ y 2022) (y 2022) ["EMPTY"] clearer (y 2023) (y 2027) Act365Fixed Following 0.025  M 12 hkd hibor12M True
+  irs4 <- originateInterestRateSwap    clearer      clearer                "IRS4"  "Interest Rate Swap 10Y 2.25%/HIBOR-12M"  pub (tt $ y 2022) (y 2022) ["EMPTY"] clearer (y 2023) (y 2032) Act365Fixed Following 0.025  M 12 hkd hibor12M True
+  cds1 <- originateCreditDefaultSwap   clearer      clearer                "CDS1"  "Credit Default Swap 10Y 4.25% TENCENT"   pub (tt $ y 2022) (y 2022) ["EMPTY"] clearer (y 2023) (y 2032) Act365Fixed Following 0.0425 M 12 hkd dpTencent rrTencent True
+  let secs = [irs1, irs2, irs3, irs4, cds1]
+
+  -- Distribution
+  buyerCashCid <- deposit operator cashProvider buyer public (qty 10_000_000.0 hkd) buyerCashAccount
+
+  -- Data
+  createCalendar clearer "EMPTY" [Saturday, Sunday] [] pub
+  createObservation clearer hibor6M [(tt $ y 2022, 0.022625), (todayTime, 0.024825)] pub
+  createObservation clearer hibor12M [(tt $ y 2022, 0.025225), (todayTime, 0.028825)] pub
+  createObservation clearer dpTencent [(tt $ y 2022, 0.0), (todayTime, 0.0)] pub
+  createObservation clearer rrTencent [(tt $ y 2022, 1.0), (todayTime, 1.0)] pub
+  createAccountDirectory buyer ([(hkd, buyerCashAccount)] <> map (\s -> (s, buyerSecAccount)) secs) $ singleton public
+  createAccountDirectory seller [(hkd, sellerCashAccount)] $ singleton public
+
+  -- Workflows
+  let
+    doTrade = do
+      buyerPaymentCid <- deposit operator cashProvider buyer public (qty 150_000.0 hkd) buyerCashAccount
+      quoteRequestCid <- requestQuote operator seller buyer (Id "RFQ-IRS1") Quoting.Buy $ qty 10_000_000.0 irs1
+      quoteCid <- quote operator seller buyer quoteRequestCid $ qty 150_000.0 hkd
```

Comment:
I think swaps are normally priced at 0 at inception (the fixed rate is adjusted to arrive at a 0 expected PV). What do these 150k HKD represent in this scenario? Maybe a transaction cost?

Reply:
Yes, I realize that. In this case we're quoting for an already-inflight swap though, so there would be an upfront fee. Admittedly, not necessarily a very realistic scenario. Typically, one would probably quote a fixed rate to a counterparty, but we currently don't support specifying these trade definitions with "holes". So the only thing we can quote on is price.

And for eg CDSs that's actually standard procedure - they're structured with a standard (eg. 5Y) duration, and standard interest payments, and are traded based on upfront fee.